common/src/secrets.rs

```diff
@@ -0,0 +1,67 @@
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::BTreeMap,
+    fmt::{Debug, Display},
+};
+use zeroize::Zeroize;
+
+/// Custom type for secrets that ensures they will not be logged accidentally and makes exposing
+/// them obvious via the [`Secret::expose`] method.
+///
+/// Also makes sure to zero out the memory of the secret when dropping.
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Secret<T: Zeroize>(T);
+
+impl<T: Zeroize> Debug for Secret<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "[REDACTED {:?}]", std::any::type_name::<T>())
+    }
+}
+
+impl<T: Zeroize> Display for Secret<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+impl<T: Zeroize> Drop for Secret<T> {
+    fn drop(&mut self) {
+        self.0.zeroize();
+    }
+}
```

Comment:
We can add unit tests for it, although I think it's pretty redundant. The `Secret` can only be displayed by using debug formatting, since it doesn't implement `Display`. The `Debug` implementation never reveals the inner value. The only way a secret could be printed is if it is accessed via `Secret::expose`, which should be obvious enough not to use accidentally.

Reply:
Might seem like paranoia, but even if the business logic looks dead simple and seems obvious in terms of the interface, I think the teams looks for a few things with the testing:
* defending against human error (modifying the source code without understanding it, modifying it for development and forgetting about it, reviewers missing to spot the changes)
* improve code quality which in its turn brings trust
* tests are seen also as specifications. `Secret` struct is sensitive because will be used across many places for secrets hiding. We want to reduce the risk and impact of an assumption being done upstream which will not be valid because changes occur to the specification.

It's also very simple to add some basic tests:
* for display & debug traits of `Secret`
* for display and debug traits of structs that use `Secret` (if possible).

Reply:
I have added some basic tests for printing `Secret` normally, with debug formatting, `.expose()` or contained inside a struct. These should make it obvious how the type is supposed to behave. Is there anything else missing?